packages/wallet/test/services/DRepRegistrationTracker.test.ts

```diff
@@ -0,0 +1,157 @@
+
+    const txWithForeignRegisterDelegateRepresentativeCertificate = {
+      body: {
+        certificates: [
+          {
+            ...registerDelegateRepresentativeCertificate,
+            dRepCredential: { hash: foreignDRepKeyHash, type: Cardano.CredentialType.KeyHash }
+          }
+        ]
+      }
+    } as unknown as Cardano.HydratedTx;
+
+    const txWithForeignUnregisterDelegateRepresentativeCertificate = {
+      body: {
+        certificates: [
+          {
+            ...unregisterDelegateRepresentativeCertificate,
+            dRepCredential: { hash: foreignDRepKeyHash, type: Cardano.CredentialType.KeyHash }
+          }
+        ]
+      }
+    } as unknown as Cardano.HydratedTx;
+
+    createTestScheduler().run(({ cold, expectObservable, flush }) => {
+      const target$ = createDRepRegistrationTracker({
+        historyTransactions$: cold('a-b-c-d-e-f-g-h', {
+          a: [],
+          b: [txWithOtherCertificate],
+          c: [txWithOtherCertificate, txWithRegisterDelegateRepresentativeCertificate_ScriptHashDRepCredential],
+          d: [
+            txWithOtherCertificate,
+            txWithRegisterDelegateRepresentativeCertificate_ScriptHashDRepCredential,
+            txWithForeignRegisterDelegateRepresentativeCertificate
+          ],
+          e: [
+            txWithOtherCertificate,
+            txWithRegisterDelegateRepresentativeCertificate_ScriptHashDRepCredential,
+            txWithForeignRegisterDelegateRepresentativeCertificate,
+            txWithOwnRegisterDelegateRepresentativeCertificate
+          ],
+          f: [
+            txWithOtherCertificate,
+            txWithRegisterDelegateRepresentativeCertificate_ScriptHashDRepCredential,
+            txWithForeignRegisterDelegateRepresentativeCertificate,
+            txWithOwnRegisterDelegateRepresentativeCertificate,
+            txWithUnregisterDelegateRepresentativeCertificate_ScriptHashDRepCredential
+          ],
+          g: [
+            txWithOtherCertificate,
+            txWithRegisterDelegateRepresentativeCertificate_ScriptHashDRepCredential,
+            txWithForeignRegisterDelegateRepresentativeCertificate,
+            txWithOwnRegisterDelegateRepresentativeCertificate,
+            txWithUnregisterDelegateRepresentativeCertificate_ScriptHashDRepCredential,
+            txWithForeignUnregisterDelegateRepresentativeCertificate
+          ],
+          h: [
+            txWithOtherCertificate,
+            txWithRegisterDelegateRepresentativeCertificate_ScriptHashDRepCredential,
+            txWithForeignRegisterDelegateRepresentativeCertificate,
+            txWithOwnRegisterDelegateRepresentativeCertificate,
+            txWithUnregisterDelegateRepresentativeCertificate_ScriptHashDRepCredential,
+            txWithForeignUnregisterDelegateRepresentativeCertificate,
+            txWithOwnUnregisterDelegateRepresentativeCertificate
+          ]
+        }),
+        pubDRepKeyHash$: of(dRepPublicKeyHash)
```

Comment:
Please add some test case(s) for when this emits `undefined`. Tip: use `hot` instead of `cold` for `historyTransactions$` to not replay all values every time `pubDRepKeyHash$` emits
```suggestion
        pubDRepKeyHash$: cold(..., { a: undefined, b: dRepPublicKeyHash })
```

Reply:
done in  [63c5f161c3dd48eef2a5e3b0225d41f662530e99](https://github.com/input-output-hk/cardano-js-sdk/pull/1192/commits/63c5f161c3dd48eef2a5e3b0225d41f662530e99)

Reply:
does this one cover it https://github.com/input-output-hk/cardano-js-sdk/pull/1192/files#diff-090a0578888402d721b446943ed78c292778aaf7f9eaef321e42cc0a2eda1091R160?

Reply:
done here https://github.com/input-output-hk/cardano-js-sdk/pull/1192/commits/c11598e6b70f4fe4e2d72d4c41f7b2856f429ac1#diff-090a0578888402d721b446943ed78c292778aaf7f9eaef321e42cc0a2eda1091R158